### PR TITLE
Fix possible IdentityProviderManagementServerException when getting IDP List

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
### Proposed changes in this pull request

This PR fixes the IdentityProviderManagementException that can occur when invoking GET idp list concurrently with another request deleting a IdP. When getting idps, the populateIdentityProviderList method iterates through existing IdPs to populate result set. If some other process is deleting an existing IDP during this process, there is a possibility of IdentityProviderManagementException.

Fixes - https://github.com/wso2-enterprise/asgardeo-product/issues/9673

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 1]
```diff
@@ -550,8 +550,12 @@ private String appendRequiredAttributes(String sqlQuery, List<String> requiredAt
             identityProvider.setImageUrl(resultSet.getString("IMAGE_URL"));
             identityProvider.setResourceId(resultSet.getString("UUID"));
 
-            populateRequiredAttributesForIdentityProviderList(resultSet, dbConnection, requiredAttributes, tenantId,
-                    identityProvider);
+            try {
+                populateRequiredAttributesForIdentityProviderList(resultSet, dbConnection, requiredAttributes, tenantId,
+                        identityProvider);
+            } catch (IdentityProviderManagementServerException e) {
```
Comment (senthalan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:556]: can't we handle this inside `populateRequiredAttributesForIdentityProviderList` ?

Comment (ChanikaRuchini) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Fixed

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 2]
```diff
@@ -680,6 +684,8 @@ private void populateRequiredAttributesForIdentityProviderList(ResultSet resultS
                     }
                 }
             }
+        } catch (IdentityProviderManagementClientException e) {
+            throw e;
```
Comment (senthalan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:688]: no need to catch and throw

Comment (ChanikaRuchini) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:688]: If not catch as IdentityProviderManagementException and throw IdentityProviderManagementServerException https://github.com/wso2/carbon-identity-framework/pull/3976/files/bf3c1b4257fca6bff8fc8476df596386891c7dda#diff-6af8c5bc4fe450a0e7aaa28ea162b84d3e862c57f60cfe438e8fd0030539ce72R689

**Review by senthalan — APPROVED** [review]

**Pull request merged**